core/include/engine/Method_MC.hpp

```diff
@@ -28,8 +28,15 @@ namespace Engine
         // Solver_Iteration represents one iteration of a certain Solver
         void Iteration() override;
 
+        // Performs a trial move of a single spin
+        bool Metropolis_Spin_Trial(int ispin, const vectorfield & spins_old, vectorfield & spins_new, const scalar & rng1, const scalar & rng2, const scalar & rng3, const scalar & cos_cone_angle);
+
+        void Block_Decomposition();
+        
         // Metropolis iteration with adaptive cone radius
         void Metropolis(const vectorfield & spins_old, vectorfield & spins_new);
+        // Parallel MC
+        void Parallel_Metropolis(const vectorfield & spins_old, vectorfield & spins_new);
```

Comment:
As a reference on the implementation (even though you didn't use it as a basis), you could reference David Bauer's diploma thesis, here or in the class comment